keps/sig-node/4817-resource-claim-device-status/README.md

```diff
@@ -0,0 +1,731 @@
+}
+
+// NetworkDeviceData provides network-related details for the allocated device.
+// This information may be filled by drivers or other components to configure
+// or identify the device within a network context.
+type NetworkDeviceData struct {
+    // InterfaceName specifies the name of the network interface associated with
+    // the allocated device. This might be the name of a physical or virtual
+    // network interface.
+    //
+    // +optional
+    InterfaceName string `json:"interfaceName,omitempty" protobuf:"bytes,1,opt,name=interfaceName"`
+
+    // Addresses lists the network addresses assigned to the device's network interface.
+    // This can include both IPv4 and IPv6 addresses.
+    // The addresses are in the CIDR notation, which includes both the address and the
+    // associated subnet mask.
+    // e.g.: "192.0.2.0/24" for IPv4 and "2001:db8::/64" for IPv6.
+    //
+    // +optional
+    // +listType=atomic
+    Addresses []string `json:"addresses,omitempty" protobuf:"bytes,2,opt,name=addresses"`
+
+    // HWAddress represents the hardware address (e.g. MAC Address) of the device's network interface.
+    //
+    // +optional
+    HWAddress string `json:"hwAddress,omitempty" protobuf:"bytes,3,opt,name=hwAddress"`
+}
+```
+
+### User Stories (Optional)
+
+#### Story 1 - Network Device Status for Network Services
+
+As a Cloud Native Network Function (CNF) vendor, my network services must be
+integrated with the network devices configured in Pods. The configuration
+properties of these network devices are therefore essential to configure the
+network services. For example, the network services must be able to route
+traffic to pods over networks attached via the network devices, the IP
+addresses of the network device(s) must then be reflected in the
+`ResourceClaim.Status` allowing the network service controller(s) to access
+them.
+
+#### Story 2 - Network Device Status for Troubleshooting
+
+As a Network Administrator, troubleshooting networking issues can be complex
+and time consuming especially when the device characteristics and operational
+status are not readily accessible. The `Devices` field in the
+`ResourceClaim.Status` provides access to comprehensive details regarding
+network interfaces helping to quickly and efficiently identify the issues such
+as error messages on failed network interface configuration, incorrect IP
+assignments or misconfigured network interfaces.
+
+### Notes/Constraints/Caveats (Optional)
+
+The content of `Data` is driver specific and not standardized as part of
+DRA, the interpretation of this field may then vary between controllers and
+users reading it. 
+
+The accuracy of the information depends on the implementation of the DRA
+Drivers, the reported status of the device may not always reflect the real time
+changes of the state of the device.
+
+### Risks and Mitigations
+
+As stated, 3rd party DRA drivers will set and update the `Devices` for
```

Comment:
So the driver itself writes this, and not Kubelet?

Reply:
yes